iwyu_tool.py

```diff
@@ -25,9 +25,67 @@
 import json
 import argparse
 import subprocess
-
-
-def run_iwyu(cwd, compile_command, iwyu_args, verbose):
+import re
+
+
+CORRECT_RE = re.compile(r'^\((.*?) has correct #includes/fwd-decls\)$')
+SHOULD_ADD_RE = re.compile(r'^(.*?) should add these lines:$')
+SHOULD_REMOVE_RE = re.compile(r'^(.*?) should remove these lines:$')
+FULL_LIST_RE = re.compile(r'The full include-list for (.*?):$')
+END_RE = re.compile(r'^---$')
+LINES_RE = re.compile(r'.*// lines ([0-9]+)-[0-9]+$')
+
+
+GENERAL, ADD, REMOVE, LIST = range(4)
```

Comment:
Just out of curiosity: What's the minimum Python version we have to support? Looking at the code, it seems to be Python 2 and 3, but which versions exactly? `namedtuple` could be used for the state tuple to make my diff a bit nicer, but that's only available for Python >= 2.6, and real enums for >= 3.4. Both are not really crucial for such short code, but anyway...